### PR TITLE
Add site ID to VCF output

### LINKED ISSUE #2103: Variant IDs in write_vcf function
Hi,
Pruning the VCF output of a tskit object using PLINK will result in PLINK errors since variant IDs are all '.'. 
Changing the195th line in 'tskit/python/tskit/vcf.py' from '.' to something simple (like pos or f'id_{pos}') temporarily solved the problem for me. I can create a pull request, but thought it would be too small of a change. 

Thanks for your great software package. 

Regards,
Roohy

### PR DESCRIPTION
**Pull request by roohy**
## Description

I added modified to python/tskit/vcf.py file to write the site ID of the variant in the ID field of the VCF file output. 
I added a test to check VCF variant ID and site ID in the ts object are the same in test_vcf.py file under the TestInterface class just in case. I did not add anything to documentation or the changelogs since I first wanted to submit this draft pull request and ask first since the change is very minimal.
Fixes #2103
 

# PR Checklist:

- [ ✅] Tests that fully cover new/changed functionality.

### CODE REVIEW

**Comment by roohy** [comment]
Hi, 
The error in the test process is raised by the mismatch between the VCF file with ID columns and the function 'legacy_write_vcf' which is implemented in the test_vcf.py to ensure compatibility with previous versions (there is also a minor Pylint issue). Is it OK if I change that function?

**Comment by benjeffery** [comment]
Thanks for the PR @roohy! Will review today.

File: python/tests/test_vcf.py [hunk 1]
```diff
@@ -473,6 +473,13 @@ def test_bad_individuals(self):
         with pytest.raises(ValueError):
             ts.write_vcf(io.StringIO(), individuals=[1, 2, ts.num_individuals])
 
+    def test_vcf_ids(self):
+        ts = msprime.simulate(2, mutation_rate=2, random_seed=1)
+        assert ts.num_sites > 0
+        with ts_to_pyvcf(ts) as vcf_reader:
+            records = zip(vcf_reader, ts.sites())
+            mismatch_list = list(filter(lambda record: (int(record[0].ID) != record[1].id), records))
+            assert len(mismatch_list) == 0
```
[the remark this answers is not in the document]
Comment (roohy) [reply python/tests/test_vcf.py:482]: I agree! I will add the changelog/docs.

**Comment by roohy** [comment]
Changelog and docs added.

**Comment by benjeffery** [comment]
> Changelog and docs added.

Great, thanks. All that is left is to fix the failing test (`tests/test_vcf.py::TestLegacyOutput`) and run the linting `pre-commit run --all`. I can do those bits if you like?

**Comment by roohy** [comment]
Thank you! I tried doing them so as not to take any more of your time. Please let me know if I am still forgetting something.

**Review by benjeffery — APPROVED** [review]
I've squashed and rebased.
Thanks for this @roohy, great to have you on the contributors list.

**Comment by roohy** [comment]
Thanks @benjeffery for your help. I hope I can make more/bigger contributions to your great software in the future!

**Pull request merged**